tests/modules/format-utils.js

```diff
@@ -188,3 +189,59 @@ test('splitString', suite => {
         });
     });
 });
+test('unescapeEntities', suite => {
+    suite.test('given non-string input - throws TypeError', t => {
+        const inputs = [
+            new Set(),
+            new Map(),
+            {},
+            true,
+            undefined,
+            0,
+            () => {},
+        ];
+        t.plan(inputs.length);
+        inputs.forEach(input => t.throws(
+            () => unescapeEntities(input),
+            TypeError,
+            `should throw for ${typeof input}`,
+        ));
+    });
+    suite.test('given empty string - returns empty string', t => {
+        const inputs = [
+            '',
+            String(),
+        ];
+        t.plan(inputs.length);
+        inputs.forEach(input => t.deepEqual(
+            unescapeEntities(input),
+            '',
+            `should return empty for empty ${input.constructor.name}`,
+        ));
+    });
+    suite.test('given 1 string - returns item as string', t => {
+        const inputs = [
+            'string',
+            'test',
+            'King\'s Arms'
+        ];
+        t.plan(inputs.length);
+        inputs.forEach(input => t.deepEqual(
+            unescapeEntities(input),
+            input,
+            'should return what it is given when no escaped entities',
+        ));
+    });
+    suite.test('given escaped entities should return unescaped string', t => {
```

Comment:
`given escaped entities - returns fully unescaped string`

Then you can combine this case and the below case with messages for inputs, e.g. "when escape is safe HTML", and "when escape is Unicode", and "when multiple escapes found".

Should add a test for what shouldn't get unescaped (like an encoded `<img src=x onerror=alert("hi")>` )

Reply:
I think that if such a string was escaped it should be decoded. But If `&lt;` was used that would not be decoded.